integration-tests/src/test/java/org/zowe/apiml/integration/ha/ChaoticTest.java

```diff
@@ -0,0 +1,118 @@
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.zowe.apiml.util.SecurityUtils.getConfiguredSslConfig;
+
+/**
+ * Verify behaviour of the application under chaotic testing
+ */
+@ChaoticHATest
+public class ChaoticTest {
+
+    private GatewayServiceConfiguration gatewayServiceConfiguration;
+    private DiscoveryServiceConfiguration discoveryServiceConfiguration;
+
+    private final String DISCOVERABLE_GREET = "/api/v1/discoverableclient/greeting";
+    private final String SHUTDOWN = "/application/shutdown";
+    final int gatewayInstances = gatewayServiceConfiguration.getInstances();
+    final int discoveryInstances = discoveryServiceConfiguration.getInstances();
+    private DiscoveryRequests discoveryRequests;
+    private String username;
+    private String password;
+    private String[] hosts;
+
+    @BeforeEach
+    void setUp() {
+        gatewayServiceConfiguration = ConfigReader.environmentConfiguration().getGatewayServiceConfiguration();
+        discoveryServiceConfiguration = ConfigReader.environmentConfiguration().getDiscoveryServiceConfiguration();
+        username = ConfigReader.environmentConfiguration().getCredentials().getUser();
+        password = ConfigReader.environmentConfiguration().getCredentials().getPassword();
+        hosts = discoveryServiceConfiguration.getHost().split(",");
+        discoveryRequests = new DiscoveryRequests(hosts[1]);
+        RestAssured.config = RestAssured.config().sslConfig(getConfiguredSslConfig());
+    }
+
+    @Nested
+    class GivenHASetUp {
+        @Nested
+        class whenOneDiscoveryServiceIsNotAvailable {
+            @Test
+            void serviceStillRegisteredToOtherDiscovery() throws URISyntaxException {
+                assumeTrue(gatewayInstances > 1 && discoveryInstances > 1);
+                String[] hosts = discoveryServiceConfiguration.getHost().split(",");
+                shutDownDiscoveryInstance(hosts[0]);
+                assertThat(discoveryRequests.isApplicationRegistered("DISCOVERABLECLIENT"), is(true));
+            }
+
+            void shutDownDiscoveryInstance(String host) throws URISyntaxException {
+                //@formatter:off
+                System.out.println(HttpRequestUtils.getUriFromDiscovery(SHUTDOWN, host));
+                given()
+                    .contentType(JSON)
+                    .auth().basic(username, password)
+                    .when()
+                    .post(HttpRequestUtils.getUriFromDiscovery(SHUTDOWN, host))
+                    .then()
+                    .statusCode(is(SC_OK))
+                    .extract().body().asString();
+                //@formatter:on
+            }
+        }
+
+        @Nested
+        class whenOneGatewayIsNotAvailable {
+            @Test
+            void routeToInstanceThroughAliveGateway() throws IOException {
+                assumeTrue(gatewayInstances > 1 && discoveryInstances > 1);
```

Comment:
Shouldn't this test verify calling via the DVIPA and therefore in this case NGINX?